### PR TITLE
Fix invalid instantiation of TemplateResponse if client not found

### LINKED ISSUE #19963: Exception: Argument 3 passed to OCP\AppFramework\Http\TemplateResponse::__construct() must be of the type array, string given, called in /var/www/nextcloud/apps/oauth2/lib/Controller/LoginRedirectorController.php on line 88
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.

### Steps to reproduce
1. Attempt to use OAuth login from Rocket Chat

### Expected behaviour
Login window appears and functions

### Actual behaviour
Window appears, says "Internal Server Error: The server was unable to complete your request. If this happens again, please send the technical details below to the server administrator. More details can be found in the server log."

Server log contains line: "Exception: Argument 3 passed to OCP\AppFramework\Http\TemplateResponse::__construct() must be of the type array, string given, called in /var/www/nextcloud/apps/oauth2/lib/Controller/LoginRedirectorController.php on line 88"

### Server configuration

**Operating system:** Ubuntu 18.04.4 LTS

**Web server:** nginx 1.16.1

**Database:** psql (PostgreSQL) 10.12 (Ubuntu 10.12-0ubuntu0.18.04.1)

**PHP version:** PHP 7.2.24-0ubuntu0.18.04.3 (cli) (built: Feb 11 2020 15:55:52) ( NTS )

**Nextcloud version:** 18.0.2

**Where did you install Nextcloud from:**
nextcloud.com

**Log**:

  | Exception: Argument  3 passed to OCP\AppFramework\Http\TemplateResponse::__construct() must  be of the type array, string given, called in  /var/www/nextcloud/apps/oauth2/lib/Controller/LoginRedirectorController.php  on line 88/var/www/nextcloud/lib/private/AppFramework/App.php - line 125: OC\AppFramework\Http\Dispatcher->dispatch(OCA\OAuth2\C ... {}, "authorize")/var/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php - line 47: OC\AppFramework\App::main("OCA\\OAuth2 ... r", "authorize", OC\AppFramew ... {}, { _route: "o ... "})<<closure>>OC\AppFramework\Routing\RouteActionHandler->__invoke({ _route: "o ... "})/var/www/nextcloud/lib/private/Route/Router.php - line 299: call_user_func(OC\AppFramew ... {}, { _route: "o ... "})/var/www/nextcloud/lib/base.php - line 1008: OC\Route\Router->match("/apps/oauth2/authorize")/var/www/nextcloud/index.php - line 38: OC::handleRequest()Caused by TypeError: Argument  3 passed to OCP\AppFramework\Http\TemplateResponse::__construct() must  be of the type array, string given, called in  /var/www/nextcloud/apps/oauth2/lib/Controller/LoginRedirectorController.php  on line 88/var/www/nextcloud/apps/oauth2/lib/Controller/LoginRedirectorController.php - line 88: OCP\AppFramework\Http\TemplateResponse->__construct("core", "404", "guest")/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 170: OCA\OAuth2\Controller\LoginRedirectorController->authorize("", "eyJsb2dpblN ... =", "code")/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 99: OC\AppFramework\Http\Dispatcher->executeController(OCA\OAuth2\C ... {}, "authorize")/var/www/nextcloud/lib/private/AppFramework/App.php - line 125: OC\AppFramework\Http\Dispatcher->dispatch(OCA\OAuth2\C ... {}, "authorize")/var/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php - line 47: OC\AppFramework\App::main("OCA\\OAuth2 ... r", "authorize", OC\AppFramew ... {}, { _route: "o ... "})<<closure>>OC\AppFramework\Routing\RouteActionHandler->__invoke({ _route: "o ... "})/var/www/nextcloud/lib/private/Route/Router.php - line 299: call_user_func(OC\AppFramew ... {}, { _route: "o ... "})/var/www/nextcloud/lib/base.php - line 1008: OC\Route\Router->match("/apps/oauth2/authorize")/var/www/nextcloud/index.php - line 38: OC::handleRequest()
-- | --

**Screenshots**:

### PR DESCRIPTION
**Pull request by kesselb**

### CODE REVIEW

File: apps/oauth2/tests/Controller/LoginRedirectorControllerTest.php [hunk 1]
```diff
@@ -24,11 +24,13 @@
 
 namespace OCA\OAuth2\Tests\Controller;
 
-use OCA\Files_Sharing\Tests\TestCase;
```
Comment (kesselb) [inline apps/oauth2/tests/Controller/LoginRedirectorControllerTest.php]: That's weird. I guess it was not intended to use `OCA\Files_Sharing\Tests\TestCase` as base class for `LoginRedirectorControllerTest`.

Comment (rullzer) [reply apps/oauth2/tests/Controller/LoginRedirectorControllerTest.php]: :see_no_evil:

**Review by rullzer — APPROVED** [review]

File: apps/oauth2/tests/Controller/LoginRedirectorControllerTest.php (hunk 1 above)
Comment (kesselb) [reply apps/oauth2/tests/Controller/LoginRedirectorControllerTest.php]: Hopefully it's unrelated but a lot of tests failed related to sharing. Restarted the test a few hours ago but job is still pending. Probably the CI want's a weekend too.

**Pull request merged**